### DISCUSSION TITLE
Check if a specific snap is installed on an user extension

### DISCUSSION THREAD

**Discussion by tcheee** [MetaMask/snaps#2307] [Q&A]
Hello,

I am looking for a way to check if the browser has metamask and a specific snap installed before showing a component in the UI or an app.

I stumbled upon [this documentation](https://docs.metamask.io/snaps/how-to/connect-to-a-snap/#determine-whether-a-snap-is-installed).

But each time I am trying to use: `method: 'wallet_getSnaps'`; I got an error code `-32601` whether the user wallet is connected or not, unlock or not.

Do you know if 'wallet_getSnaps' is desprecated? And what is the current best solution to check the snaps installed on a user wallet?

Thank you!

**Comment by Montoya** [comment]
`wallet_getSnaps` only works if the user has connected the Snap to the requesting dapp. If a user installs a Snap `npm:coolsnap` on `site1.com` and then `site2.com` calls `wallet_getSnaps` with the id `npm:coolsnap`, it will return false even though the user has it installed, because `site2.com` does not have permission to talk to `npm:coolsnap` and thus cannot even ask if it is installed. You have to do `wallet_requestSnaps` with `npm:coolsnap` first. Does that help?
